From cd4210cbdf8e3efc95368285a3d75af4b2eff71f Mon Sep 17 00:00:00 2001
From: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
Date: Fri, 14 Feb 2020 16:47:25 +0530
Subject: [PATCH 1207/1851] firmware: xilinx: Add OSPI MUX selection support

commit 405bb5c27be4faafd69bd27c12a412cd5d9f839f from
https://github.com/Xilinx/linux-xlnx.git

This patch adds OSPI Mux selection node to select between
DMA and Linear interface.

Signed-off-by: Sai Krishna Potthuri <lakshmi.sai.krishna.potthuri@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp.c     | 1 +
 include/linux/firmware/xlnx-zynqmp.h | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index 6e21b57a5181..4f2323bc2948 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -529,6 +529,7 @@ static inline int versal_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_PROBE_COUNTER_READ:
 	case IOCTL_PROBE_COUNTER_WRITE:
 	case IOCTL_USB_SET_STATE:
+	case IOCTL_OSPI_MUX_SELECT:
 		return 1;
 	default:
 		return 0;
diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index 804c24d1fae1..2431e2c7adf3 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -458,6 +458,12 @@ enum pm_register_access_id {
 	CONFIG_REG_READ,
 };
 
+enum ospi_mux_select_type {
+	PM_OSPI_MUX_SEL_DMA,
+	PM_OSPI_MUX_SEL_LINEAR,
+	PM_OSPI_MUX_GET_MODE,
+};
+
 enum pm_node_id {
 	NODE_UNKNOWN = 0,
 	NODE_APU,
-- 
2.31.1

